scripts/test-summary.sh

```
@@ -0,0 +1,360 @@
+    
+    # Extract numbers for calculations
+    local total_suites=$(echo "$suite_stats" | grep -oE '[0-9]+ total' | grep -oE '[0-9]+')
+    local failed_suites=$(echo "$suite_stats" | grep -oE '[0-9]+ failed' | grep -oE '[0-9]+' || echo "0")
+    local passed_suites=$(echo "$suite_stats" | grep -oE '[0-9]+ passed' | grep -oE '[0-9]+' || echo "0")
+    
+    local total_tests=$(echo "$test_stats" | grep -oE '[0-9]+ total' | grep -oE '[0-9]+')
+    local failed_tests=$(echo "$test_stats" | grep -oE '[0-9]+ failed' | grep -oE '[0-9]+' || echo "0")
+    local passed_tests=$(echo "$test_stats" | grep -oE '[0-9]+ passed' | grep -oE '[0-9]+' || echo "0")
+    
+    # Calculate pass rates
+    local suite_pass_rate=$(awk "BEGIN {printf \"%.1f\", ($passed_suites / $total_suites) * 100}")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)